### ISSUE TITLE
Initial acceptance testing

### ISSUE THREAD

**Issue by ScalaWilliam** [ActionFPS/ActionFPS-Game#98]
Hi guys, 

We need some of you to test the new game binaries.
Game downloads are here:
https://github.com/ActionFPS/ActionFPS-Game/releases/tag/circle_jq

When you have installed the game, go to https://actionfps.com/play/ and click the authentication link at the bottom.

<img width="392" alt="screen shot 2017-02-01 at 08 52 22" src="https://cloud.githubusercontent.com/assets/2464813/22491050/c8cb7a60-e85b-11e6-95ae-1ba1e2fa9028.png">

Then it'll open the game, and you should be able to type `/connect woop.ac 7654` and the connect should just work.

Any issues, all post here. We'll need exact reproducible steps.

Regards,

**Comment by DaylixX** [comment]
For me its bad commandline syntax ><

**Comment by lucasgautheron** [comment]
Post the link here.

**Comment by DaylixX** [comment]
@lucasgautheron its good i deleted a last version of Match Client =)

**Comment by lucasgautheron** [comment]
So this is working for you, no issue to report ?

**Comment by DaylixX** [comment]
Is it normal if I don't have a server in the list ?

**Comment by DaylixX** [comment]
Ok server `connect woop.ac 7654` its working now

**Comment by Paul-Michaud** [comment]
I had to restart the game after I clicked the button "Click here to connect up automatically." to be able to connect to woop.ac 7654

I had the SmartScreen warning when installing (Same as the normal AC I think)

Otherwise everything seems to be working !

**Comment by lucasgautheron** [comment]
"I had to restart the game after I clicked the button "Click here to connect up automatically." to be able to connect to woop.ac 7654"
What happened exactly otherwise ?

**Comment by Paul-Michaud** [comment]
1. Installed in my document
2. Started & closed the game
3. Clicked the link on https://actionfps.com/play/, it started the game 
4. Typed  /connect woop.ac 7654 but it did nothing, closed the game
5. Re-clicked the link, same problem
6. Started the game normally (with the shortcut on my desktop), it worked

**Comment by lucasgautheron** [comment]
* "Installed in my document" I don't understand if you're talking about the installation mode (single user vs shared) or location ? Knowing both might help
 * did nothing = it didn't even attempt to connect at all ?
 * Can you make sure clicking a link starts the right version of AC ? It seems so me like this is the issue actually. One way to check that in game is to take a screenshot and check the output path.
Maybe trying uninstalling previous versions of the matchclient that might rely on the actionfps:// protocol before and reinstalling actionfps.

**Comment by Paul-Michaud** [comment]
I picked the second option 
![image](https://cloud.githubusercontent.com/assets/7859727/22517810/ae8f406a-e8aa-11e6-8862-ffdb67233e80.png)
Then
![image](https://cloud.githubusercontent.com/assets/7859727/22517835/c4ff188e-e8aa-11e6-8899-b3e8b17bd3d5.png)

did nothing = it didn't even attempt to connect at all ? -> Yes

And yes, it started the right version of AC

**Comment by lucasgautheron** [comment]
@Paul-Michaud 
> And yes, it started the right version of AC

How did you check that ?

**Comment by Paul-Michaud** [comment]
![image](https://cloud.githubusercontent.com/assets/7859727/22518192/0a46c06c-e8ac-11e6-8316-4b5204874344.png)

**Comment by lucasgautheron** [comment]
Alright! Thanks.

**Comment by lucasgautheron** [comment]
@ketarXVII we have yet to get your feedback, mac testing is precious and this should really not take long...

**Comment by ketarXVII** [comment]
I got it ty. I'll test it

**Comment by ScalaWilliam** [comment]
PS for OSX I think this is relevant https://github.com/ActionFPS/ActionFPS-Game/issues/94

**Comment by ketarXVII** [comment]
Mac version doesn't open... it opens just the launcher and when I say PLAY nothing happens. Tried to change window size and all, but nothing works

**Comment by ScalaWilliam** [comment]
What do you mean 'say PLAY'?

**Comment by ScalaWilliam** [comment]
Can you go to an app called 'Console' and find if there's any log messages?

And also output of `cat ~/Library/Application\ Support/actionfps/next/clientlog.txt`

**Comment by ketarXVII** [comment]
sorry. When I press the red Play button in the launcher (i.e. the first small window where you set window size and other options). Thought it was obvious.

**Comment by ketarXVII** [comment]
com.apple.message.domain: com.apple.crashreporter.writereport.crash
com.apple.message.signature: actionfps
com.apple.message.signature2: UNBUNDLED ||| ??? (1.2.x.x)
com.apple.message.signature3: UNKNOWN
com.apple.message.result: NO
com.apple.message.summarize: YES
SenderMachUUID: E09A71B1-7DB9-3F89-8E54-164DE703B561

**Comment by ScalaWilliam** [comment]
Anything AF related though?

**Comment by ScalaWilliam** [comment]
Can you provide a screenshot when you click Apple icon in top left and then 'About'?

**Comment by lucasgautheron** [comment]
Would like to know which Mac OS X version as well :)

**Comment by ketarXVII** [comment]
wait is this useful?

**Comment by ketarXVII** [comment]
Process:               actionfps [3019]
Path:                  /Users/USER/Desktop/actionfps.app/Contents/gamedata/actionfps.app/Contents/MacOS/actionfps
Identifier:            actionfps
Version:               1.2.x.x
Code Type:             X86-64 (Native)
Parent Process:        ??? [3009]
Responsible:           actionfps [3019]
User ID:               502

Date/Time:             2017-02-02 16:41:18.810 +0100
OS Version:            Mac OS X 10.12.2 (16C67)
Report Version:        12
Anonymous UUID:        D6FB7EEE-B2EB-5854-5EB4-2F7C2CA804DF


Time Awake Since Boot: 19000 seconds

System Integrity Protection: enabled

Crashed Thread:        0

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    DYLD, [0x1] Library missing

Application Specific Information:
dyld: launch, loading dependent libraries

Dyld Error Message:
  Library not loaded: /usr/local/opt/openssl/lib/libssl.1.0.0.dylib
  Referenced from: /Users/USER/Desktop/actionfps.app/Contents/gamedata/actionfps.app/Contents/MacOS/actionfps
  Reason: image not found

Binary Images:
       0x106188000 -        0x1062e6ff7 +net.cubers (1.2.x.x) <0B75530B-1A45-3023-A2A6-6049D059F5FC> /Users/USER/Desktop/actionfps.app/Contents/gamedata/actionfps.app/Contents/MacOS/actionfps
       0x108be1000 -        0x108c1e267  dyld (421.2) <947FC440-80F9-32F7-A773-6FC418FE1AB7> /usr/lib/dyld

**Comment by ketarXVII** [comment]
version is sierra 10.12.2

**Comment by lucasgautheron** [comment]
Yes this is very useful

**Comment by ScalaWilliam** [comment]
Would you try `brew install openssl` as a temporary fix? Brew is: http://brew.sh/

Looks like it linked dynamically and not statically against this lib.

I'm on 10.11 myself but that shouldn't make a difference afaik.

**Comment by ScalaWilliam** [comment]
Yes all of it is useful :) Where and how did you get this information? "Dyld Error Message" etc.

**Comment by ketarXVII** [comment]
from console- user reports (I guess in English, mine is in ITA)

**Comment by ScalaWilliam** [comment]
try brew fix, I'll try to fix problem for new users soon

**Comment by ketarXVII** [comment]
that brew shit is safe right? I do not want to mess things up

**Comment by ketarXVII** [comment]
this is a production machine, can't risk

**Comment by ScalaWilliam** [comment]
http://apple.stackexchange.com/questions/241758/is-homebrew-safe-for-mac
https://www.quora.com/Is-it-safe-to-install-either-Homebrew-or-MacPorts-on-OS-X

**Comment by ketarXVII** [comment]
"It's safe, if you know what you're downloading." and i do not know. :) In case you are in charge

**Comment by ScalaWilliam** [comment]
openssl, which we use for authentication. Alternatively, you can wait a bit until I fix the problem and package openssl with af.

**Comment by ketarXVII** [comment]
brew installed. now?

**Comment by ScalaWilliam** [comment]
`brew install openssl` & try running everything again..

**Comment by ketarXVII** [comment]
sry for being noobie, how to install openssl? copied that command - if it is a command - in console but no output. You have to guide me through the process

**Comment by ketarXVII** [comment]
I thought the cmmand was brew install openssl iMac-di-xxxx:~ xxxxxx$ install openssl
usage: install [-bCcpSsv] [-B suffix] [-f flags] [-g group] [-m mode]
               [-o owner] file1 file2
       install [-bCcpSsv] [-B suffix] [-f flags] [-g group] [-m mode]
               [-o owner] file1 ... fileN directory
       install -d [-v] [-g group] [-m mode] [-o owner] directory ...

**Comment by lucasgautheron** [comment]
It should be exactly:
```brew install openssl```

**Comment by ketarXVII** [comment]
iMac-di-XXXX:~ XXXXXX$ brew install openssl
-bash: brew: command not found

**Comment by lucasgautheron** [comment]
You need to install homebrew

**Comment by ketarXVII** [comment]
I did that

**Comment by lucasgautheron** [comment]
```-bash: brew: command not found``` means it was not installed properly. maybe close and open your terminal again.

**Comment by ketarXVII** [comment]
yeah, I re-installed it, Probably there was some error I did not notice in the first installation. Now the command brew install openssl worked. Launching the client now works but the window it shows is black and weird. Nice AC music in the background! 
![schermata 2017-02-02 alle 17 26 13](https://cloud.githubusercontent.com/assets/13039862/22558135/f2207eee-e96c-11e6-9c57-fcf7ba86c049.png)

**Comment by lucasgautheron** [comment]
Apparently @RonaldReagan is having a hard time getting AC to work on all mac versions. I have no idea if your version should be considered as outdated or not, if SDL 1.2 could have something to do with this, etc.
At first sight, it looks like there are two issues:
 * Textures not loading?
 * No light

**Comment by ketarXVII** [comment]
my version is the latest. 
Hard to say what is missing, it is all black. hitting the mouse button produce some results (the outlined window change measures, like going through the menu) and finally I can shoot. In the dark.

**Comment by lucasgautheron** [comment]
what if you tweak the settings in the launcher? does any combination solve these issues?

**Comment by ketarXVII** [comment]
no, not rly. Launching it more times while tweaking FSAA in a couple of occasions I could see something (a door), barely in the dark. Probably it depends on where I spawn. So yes, there is certainly a light issue. Maybe textures are loaded. Hard to say.

**Comment by lucasgautheron** [comment]
I meant the mac launcher settings. I have no idea what it is and what is looks like, but you mentioned a launcher. I think it has the ability to turn on and off fullscreen maybe, I don't know.

**Comment by ketarXVII** [comment]
Yes. This is the Mac launcher. Exactly what you guessed 
![schermata 2017-02-02 alle 17 55 18](https://cloud.githubusercontent.com/assets/13039862/22559247/d3712008-e970-11e6-88aa-7a744c1fc251.png)

**Comment by lucasgautheron** [comment]
What matters the most is fullscreen vs windowed i think :) try both.

**Comment by ketarXVII** [comment]
of course I did that, I tried any combination, nothing changes

**Comment by lucasgautheron** [comment]
Ok thanks. We need a mac guy to find out what's the problem.

**Comment by ScalaWilliam** [comment]
I'll upgrade one of my macs to 10.12, see why this is. Looks screwed up!

**Comment by ScalaWilliam** [comment]
Thanks for the help Ketar :)

**Comment by ScalaWilliam** [comment]
I upgraded to 10.12, can confirm the issue too.

**Comment by ScalaWilliam** [comment]
#123 fails the testing.

**Comment by lucasgautheron** [comment]
@ketarXVII

**Comment by ketarXVII** [comment]
yes yes yes

**Comment by ketarXVII** [comment]
do I have to dload the new client?

**Comment by ScalaWilliam** [comment]
link coming

**Comment by ScalaWilliam** [comment]
https://234-47462659-gh.circle-artifacts.com/0/Users/distiller/ActionFPS-Game/source/xcode/build/Release/actionfps.dmg

**Comment by ketarXVII** [comment]
ok it opens, now?

**Comment by lucasgautheron** [comment]
https://actionfps.com/play/

**Comment by ketarXVII** [comment]
first bug: opening it windowed doesn't allow me to switch to another window. To write here I had to quit it

**Comment by lucasgautheron** [comment]
yep we found this bug earlier
are you coming?

**Comment by ketarXVII** [comment]
yes but auth failed

**Comment by lucasgautheron** [comment]
have you clicked the red button?
it would be interested to have AC logs as well (everything that has been printed in the console)

**Comment by lucasgautheron** [comment]
@ketarXVII can you post github issues for each problem you've met ?

**Comment by francoisleduc** [comment]
Using Ubuntu 16.04 LTS - 64Bits 


- The installation went very well, except that it was missing a dependency since you made the "libsdl2 merge" (so it should be added in the dependencies on the site -> sudo apt-get install libsdl2-image- 2.0-0) 

- Once the installation was finished and I had changed some settings, I quit the game to try and start it from the red "Play now (with auth)" button. It didn't work, instead of launching the game it showed a white page in my browser saying:

"Firefox can not open this address because one of the following protocols (actionfps) is not associated with any program or is not allowed in this context.
It may be necessary to install another application to open this type of address." **(NOT OK)**

_Shouldn't be caused by the browser i guess, but note that i was not using Google Chrome but I was correctly registered on the actionfps web site._

- I still wanted to try if I could connect to aura.woop.us 7654 without any auth key but it refused me the access. This part seems to work well. **(OK)**

Here are some logs from the console.

https://pastebin.com/jirUzdCu   <- logs  (few "Could not read" errors)

Bugs I've met in during my short test session: 

1. 
Editing gamma by using the command or the player menu/settings/gamma for this map/ doesn't work. (Keep in mind that it doesn't work in the Official Assault Cube released version either). I even tried /recalc , didn't work (I don't know if it should tho).

https://pastebin.com/gSEVa1Ce

Once I clicked "Apply now" it refreshes my game window but it also resets the gamma to 100. (default one)

2.
I also noticed that /lightscale x wasn't recognized by the game. It recognizes it in a way that it appears in white and suggests me the values (min-max range) but when I press enter to set it to a certain value it doesn't recognizes it (_showed in first log pastebin_).  Seems a little bit weird because it should.

I haven't tried any other commands in the game but some other ones might not work too.

3. 

It seems like in editing mode (pressing "e" key) all my keyboard settings are forgotten. I couldn't go left with Q anymore and go forward with Z.  I was using  a French keyboard and had to make these changes to play correctly.

**Comment by lucasgautheron** [comment]
Hello @francoisleduc !
Thank you very much for your input. First of, have you run the installation script (./install.sh) before trying to launch the game ?

**Comment by francoisleduc** [comment]
Here we go !
I don't remember if I forgot  something the first time but it worked by re-executing ./install.sh

jesus@hosta:~/Documents/ActionFPS$ sudo ./install.sh
[sudo] Mot de passe de jesus : 
The following menuitem(s) currently exist:
/home/jesus/.local/share/applications/actionfps.desktop
Would you like them all to be deleted? (y/N): N
The existing menuitems will remain.

Registering xdg-mime
An ActionFPS menuitem has been successfully created at
/home/jesus/.local/share/applications/actionfps.desktop

Worked perfectly fine when I launched the game from the "Play now (with auth)" button

<a href="http://www.casimages.com/i/170422123746604634.png.html" target="_blank" title="goodActionsfps"><img src="http://nsa37.casimages.com/img/2017/04/22/170422123746604634.png" border="0" alt="goodActionsfps" /></a>


And finally 

<a href="http://www.casimages.com/i/170422123746696518.jpg.html" target="_blank" title="actionsfps"><img src="http://nsa37.casimages.com/img/2017/04/22/170422123746696518.jpg" border="0" alt="actionsfps" /></a>

Feel free to ask me to try different things on the client.

**Comment by mazellanse** [comment]
Hi, I installed actionfps on my Ubuntu14.04. I installed the suggested packages. If I remember correctly I had to install libsdl1.2debian manually because libsdl2 was already installed and the system then thought that it already had the latest of that. 

After that I ran the installation. I had to do that with sudo since it wanted to write to /usr/share/mime/packages/application-vnd.actionfps-dmo+gz. Looked like all went fine.

If I then go to https://actionfps.com/play/ and klick the "play now" button it wants me to confirm to open the link with actionfps. When I do that the resolution on the screen changes to something like 600x400 or less. I'm still looking at the webpage but nothing else happens. In order to get out of that low-res mode I have to restart the computer :s 

Any ideas?

/Mazellan

**Comment by francoisleduc** [comment]
Seems unfortunate, no idea where that bug might come from. I also had to install/re-install libsdl2 , it may come from there.

I suppose that you tried changing your screen resolution in your system settings before rebooting your computer, right :) ?

**Comment by ScalaWilliam** [comment]
@mazellanse thanks - I had a similar problem before, your Ubuntu is quite old. IIRC you can use `xrandr` to fix this by changing your resolution: https://wiki.ubuntu.com/X/Config/Resolution


Do you want to try `/screenres <your native res>` as well when in ActionFPS, maybe the resolution will be reset like that too.
